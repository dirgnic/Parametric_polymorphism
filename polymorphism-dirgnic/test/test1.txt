def add[A](x : A, y : A) {
  return x;
}

def main() {
  return add[Int](10, 20);
}
