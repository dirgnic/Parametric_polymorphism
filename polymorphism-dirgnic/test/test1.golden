=== After Monomorphization ===
Def "main" [] [] (App (App (Var "add_TInt") (IntLit 10)) (IntLit 20))
Def "add_TInt" [] [("x",TInt),("y",TInt)] (Var "x")
Num 10
